packages/fxa-content-server/app/scripts/views/mixins/email-autocomplete-domains-mixin.js

```diff
@@ -0,0 +1,124 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A mixin to autocomplete common domains from `datalist option` `value`s.
+ */
+import Template from 'templates/partial/email-autocomplete-domains.mustache';
+import UserAgentMixin from '../../lib/user-agent-mixin';
+
+export const EMAIL_SELECTOR = 'input[type=email]';
+export const DATALIST_OPTIONS_SELECTOR = '#autocomplete-domain option';
+export const FOCUS_HACK_SELECTOR = '#focus-hack';
+
+export const DOMAINS = [
+  'gmail.com',
+  'outlook.com',
+  'hotmail.com',
+  'yahoo.com',
+  'qq.com',
+  'web.de',
+  'aol.com',
+  'mail.ru',
+  'icloud.com',
+  'gmx.de',
+  't-online.de',
+  'orange.fr',
+  'yandex.ru',
+  'yahoo.fr',
+  'live.com',
+  '163.com',
+  'msn.com',
+  'comcast.net',
+  'hotmail.co.uk',
+  'hotmail.fr',
+];
+
+export default {
+  events: {
+    [`keyup ${EMAIL_SELECTOR}`]: '_toggleDomainAutocomplete',
```

Comment:
Have you tested this with Firefox's username/password fill feature to make sure they interact well together? With the normal form fill, I see a little bit of jankiness when typing/erasing the `@`, but nothing to block on:

![flashy-email-dropdown](https://user-images.githubusercontent.com/848085/69434121-e898a600-0d34-11ea-9e20-f09cee3ef9df.gif)

Reply:
Just when you think you've tested a ton of edge cases😂 It looks like when there's an autocomplete list along with the `datalist` set and you play with backspacing/reenter the "@" sign, there's that jankiness. I'm seeing it on Chrome too, it's got to be around the `focusHack` firing when it probably shouldn't be.

Reply:
This isn't a blocker, I wouldn't spend too much time on it.

Reply:
@shane-tomlinson I'm going to ask QA to vet this thing and file an issue with all the findings. At the very least, we should have a list of all of any awkwardness/edge cases like this and determine what we can fix or what should be changed. I can't find a quick solution for this one in particular because the autocomplete drop-down list is not in the DOM.

Other changes have been pushed.

Reply:
(I am, however, going to test in Edge real quick to make sure nothing is broken)